tools/nixpkgs-hammer

```diff
@@ -214,80 +312,82 @@ def main(args):
                 overlays = {indent(overlays_nix, 2 + 2).strip()};
             }};
         in {indent(attr_messages_nix, 2 + 0).strip()}
-        '''
+        """
     )
 
     if args.show_trace:
-        print('Nix expression:', all_messages_nix, file=sys.stderr)
+        print("Nix expression:", all_messages_nix, file=sys.stderr)
 
     overlay_data = nix_eval_json(all_messages_nix, args.show_trace)
-    all_overlay_messages = {
-        attr_path: value["report"]
-        for attr_path, value in overlay_data.items()
-    }
-    attr_files = {
-        attr_path: value["position"]
-        for attr_path, value in overlay_data.items()
-        if value['position'] is not None
-    }
-
-    all_ast_check_messages = ast_checks(attr_files, args.excluded_rules)
-    all_messages = concatenate_messages(all_overlay_messages, all_ast_check_messages)
+    overlay_reports = {attr.name: reports for attr, reports in overlay_data.items()}
+    external_reports = run_external_checks(
+        list(overlay_data.keys()), args.excluded_rules
+    )
+    all_messages = concatenate_messages(overlay_reports, external_reports)
 
     if args.json:
-        print(json.dumps(all_messages))
+        print(json.dumps(all_messages, cls=JSONEncoder))
     else:
-        for attr, messages in all_messages.items():
-            print(bold(f'When evaluating attribute ‘{attr}’:'), file=sys.stderr)
+        for attr_name, messages in all_messages.items():
+            # print(messages)
+            print(bold(f"When evaluating attribute ‘{attr_name}’:"), file=sys.stderr)
             if len(messages) > 0:
-                print('\n'.join(map(lambda msg: stringify_message(**msg), messages)), file=sys.stderr)
+                print(
+                    "\n".join([message.stringify() for message in messages]),
+                    file=sys.stderr,
+                )
             else:
-                print(green('No issues found.'), file=sys.stderr)
+                print(green("No issues found."), file=sys.stderr)
             print(file=sys.stderr)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```

Comment:
This is quite hard to review. Could you please split the style changes and ast → rust rename into separate commits and rebase the PR onto them so that they do not overshadow the changes?

Reply:
Will do.